### PR TITLE
feat(RELEASE-1047): pass SA to pipelineRuns

### PR DESCRIPTION
**Pull request by johnbieren**
This commit modifies the adapter to pass the release-service-account service account to the pipelineRuns created by the adapter.

### CODE REVIEW

File: api/v1alpha1/internalrequest_types.go [hunk 1]
```diff
@@ -42,6 +42,12 @@ type InternalRequestSpec struct {
 	// Timeouts defines the different Timeouts to use in the InternalRequest PipelineRun execution
 	// +optional
 	Timeouts tektonv1beta1.TimeoutFields `json:"timeouts,omitempty"`
+
+	// ServiceAccount defines the serviceAccount to use in the InternalRequest PipelineRun execution.
+	// If none is passed, the default pipeline one will be used
```
Comment (davidmogar) [inline api/v1alpha1/internalrequest_types.go:47]:
```suggestion
	// If none is passed, the default Tekton ServiceAccount will be used
```

File: tekton/pipeline_run.go [hunk 2]
```diff
@@ -93,6 +93,10 @@ func (i *InternalRequestPipelineRun) WithInternalRequest(internalRequest *v1alph
 		i.Spec.Timeouts = &internalRequest.Spec.Timeouts
 	}
 
+	if internalRequest.Spec.ServiceAccount != "" {
```
Comment (davidmogar) [inline tekton/pipeline_run.go:96]: You can assign it directly. Empty string is the default value so it won't get added.

File: tekton/pipeline_run_test.go [hunk 3]
```diff
@@ -106,6 +107,13 @@ var _ = Describe("PipelineRun", Ordered, func() {
 
 			Expect(newInternalRequestPipelineRun.Spec.Timeouts).To(Equal(timeouts))
 		})
+
+		It("should set the ServiceAccountName for the PipelineRun", func() {
```
Comment (davidmogar) [inline tekton/pipeline_run_test.go:111]: Maybe a test to ensure it's not added if not defined?

**Review by davidmogar — APPROVED** [review]

File: tekton/pipeline_run_test.go (hunk 3 above)
Comment (johnbieren) [reply tekton/pipeline_run_test.go:111]: added

File: tekton/pipeline_run.go (hunk 2 above)
Comment (johnbieren) [reply tekton/pipeline_run.go]: fixed

File: api/v1alpha1/internalrequest_types.go (hunk 1 above)
Comment (johnbieren) [reply api/v1alpha1/internalrequest_types.go]: fixed

**Pull request merged**